### PR TITLE
#126 [Contribution] [Content] [Plugin] + add plugin '+' on contentset

### PR DESCRIPTION
**Pull request by fkroockmann**
This plugin add content directly in the contentset.
When contentset don't have accept, i will show a popin for choose a content.

### CODE REVIEW

**Comment by ndufreche** [comment]
:+1:

File: src/tb/apps/content/plugins/contentsetplus/main.js [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            },
+
+            /**
+             * Show popin and bind events
+             * @param {Mixed} contents
+             */
+            showPopin: function (contents) {
+                var config = {};
+
+                if (this.widget === undefined) {
```
Comment (mickaelandrieu) [inline src/tb/apps/content/plugins/contentsetplus/main.js]:
```js
if( typeof this.widget === "undefined") {
[...]
}
```

File: src/tb/apps/content/plugins/contentsetplus/main.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            /**
+             * Verify if the plugin can be apply on the context
+             * @returns {Boolean}
+             */
+            canApplyOnContext: function () {
+                return true;
+            },
+
+            /**
+             *
```
Comment (mickaelandrieu) [inline src/tb/apps/content/plugins/contentsetplus/main.js]: remove unused line :)

File: src/tb/apps/content/plugins/contentsetplus/main.js (hunk 1 above)
Comment (ndufreche) [inline src/tb/apps/content/plugins/contentsetplus/main.js]: there is no differences, just one are short than the other.

File: src/tb/apps/content/repository/content.repository.js [hunk 3]
```diff
@@ -59,8 +60,19 @@ define(
                  * @returns {Promise}
                  */
                 findCategories: function () {
-                    return CoreDriverHandler.read(this.TYPE + '-category');
+                    var self = this,
+                        dfd = jQuery.Deferred();
+
+                    if (this.categories === undefined) {
+                        CoreDriverHandler.read(this.TYPE + '-category').done(function (categories) {
```
Comment (mickaelandrieu) [inline src/tb/apps/content/repository/content.repository.js]: this.TYPE or this.type ?

File: src/tb/apps/content/templates/palette-blocks.twig [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<div id="palette-blocks" class="bb5-dialog ui-dialog-no-title">
```
Comment (mickaelandrieu) [inline src/tb/apps/content/templates/palette-blocks.twig]: why do we have twig files in BbCoreJS ? use Nunjucks templates instead (quite same syntax).

File: src/tb/apps/content/plugins/contentsetplus/main.js (hunk 1 above)
Comment (mickaelandrieu) [inline src/tb/apps/content/plugins/contentsetplus/main.js]: there is dude :)

```js
typeof undefined_variable
"undefined"
undefined === undefined_variable
VM268:2 Uncaught ReferenceError: undefined_variable is not defined
```

**Comment by mickaelandrieu** [comment]
this is good work, with very clean and nice javascript. GG bro' ;)

File: src/tb/apps/content/plugins/contentsetplus/main.js (hunk 1 above)
Comment (ndufreche) [inline src/tb/apps/content/plugins/contentsetplus/main.js]: Yes sure, but

```javascript
var foo = {};
foo.bar === undefiend;
>>> true
```
And we are in this case :sunglasses:

**Comment by fkroockmann** [comment]
Thanks, ready for merge :)

**Pull request merged**